### PR TITLE
Each modifications group now have a dedicated application status

### PR DESCRIPTION
**Pull request by Meklo**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/dto/NetworkModificationResult.java [hunk 1]
```diff
@@ -11,9 +11,7 @@
 import lombok.Data;
 import org.gridsuite.modification.server.impacts.SimpleElementImpact;
 
-import java.util.List;
-import java.util.Set;
-import java.util.TreeSet;
+import java.util.*;
```
Comment (Walid-Sahnoun) [inline src/main/java/org/gridsuite/modification/server/dto/NetworkModificationResult.java:14]: no need to import the whole library

File: src/test/java/org/gridsuite/modification/server/Impacts/ElementImpactTest.java [hunk 2]
```diff
@@ -16,10 +16,7 @@
 import org.gridsuite.modification.server.utils.TestUtils;
 import org.junit.Test;
 
-import java.util.Collection;
-import java.util.HashSet;
-import java.util.List;
-import java.util.TreeSet;
+import java.util.*;
```
Comment (Walid-Sahnoun) [inline src/test/java/org/gridsuite/modification/server/Impacts/ElementImpactTest.java:19]: same here

File: src/main/java/org/gridsuite/modification/server/dto/NetworkModificationResult.java [hunk 3]
```diff
@@ -39,9 +37,12 @@ public ApplicationStatus max(ApplicationStatus other) {
         }
     }
 
-    @Schema(description = "Application status")
+    @Schema(description = "Operation application status")
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/dto/NetworkModificationResult.java:40]: Modification application status or Network modification application status ?

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/dto/NetworkModificationResult.java [hunk 4]
```diff
@@ -39,9 +37,12 @@ public ApplicationStatus max(ApplicationStatus other) {
...
     ApplicationStatus applicationStatus;
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/dto/NetworkModificationResult.java:41]: Remove it

File: src/main/java/org/gridsuite/modification/server/modifications/NetworkModificationApplicator.java [hunk 5]
```diff
@@ -56,36 +57,45 @@ public NetworkModificationApplicator(NetworkStoreService networkStoreService, Eq
...
         return listener.flushNetworkModifications();
     }
 
     public NetworkModificationResult applyModifications(List<Pair<String, List<ModificationInfos>>> modificationInfosGroups, NetworkInfos networkInfos, UUID reportUuid) {
         NetworkStoreListener listener = NetworkStoreListener.create(networkInfos.getNetwork(), networkInfos.getNetworkUuuid(), networkStoreService, equipmentInfosService);
-        modificationInfosGroups.forEach(g -> g.getRight().forEach(m -> apply(m, listener, new ReportInfos(reportUuid, g.getLeft()))));
+        List<ApplicationStatus> groupsApplicationStatuses =
+                modificationInfosGroups.stream()
+                .map(g -> apply(g.getRight(), listener.getNetwork(), new ReportInfos(reportUuid, g.getLeft())))
+                .collect(Collectors.toList());
```
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/modification/server/modifications/NetworkModificationApplicator.java:73]: `.collect(Collectors.toList());` -> `toList();`

Comment (Meklo) [reply src/main/java/org/gridsuite/modification/server/modifications/NetworkModificationApplicator.java]: Done

File: src/test/java/org/gridsuite/modification/server/Impacts/ElementImpactTest.java (hunk 2 above)
Comment (Meklo) [reply src/test/java/org/gridsuite/modification/server/Impacts/ElementImpactTest.java]: Done

File: src/main/java/org/gridsuite/modification/server/dto/NetworkModificationResult.java (hunk 1 above)
Comment (Meklo) [reply src/main/java/org/gridsuite/modification/server/dto/NetworkModificationResult.java]: Done

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:1]: the field reportService is no longer used.

**Review by AbdelHedhili — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java
Comment (Meklo) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:1]: I removed it

**Review by AbdelHedhili — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Comment by SlimaneAmar** [comment]
Review OK
Test OK

**Review by Walid-Sahnoun — APPROVED** [review]
code ok
test ok

**Pull request merged**